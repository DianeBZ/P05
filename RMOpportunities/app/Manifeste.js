import React, { Component } from 'react';
import { Text, Image, StyleSheet, View, ScrollView, Dimensions, TouchableHighlight, BackAndroid, Navigator,} from 'react-native';

import Translation from './Translation';
import SigningUp from './SigningUp';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

export default class Manifeste extends Component{
 render(){
		 BackAndroid.addEventListener('Back',this.onBackAndroid);
		 return(
				<ScrollView style={styles.scrollView}>
					<Text style={styles.titleManifeste}>{Translation[lang].manifeste}</Text>
					<Text style={styles.manifesteText}>{Translation[lang].manifesteText}</Text>
					<TouchableHighlight onPress={this._onPressRegister}>
						<Text style={styles.manifesteRegister}>{Translation[lang].manifesteRegister}</Text>
					</TouchableHighlight>
					<Text />
					<Text />
				</ScrollView>
				);
		}
 onBackAndroid = () => {
	const { navigator } = this.props;
	if (navigator && navigator.getCurrentRoutes().length > 1) {
		BackAndroid.removeEventListener('Back', this.onBackAndroid);
		navigator.pop();
		return true;
	} else {
		return false;
	}
 };
 
  _onPressRegister = () => {
	BackAndroid.removeEventListener('Back', this.onBackAndroid);
	const { navigator } = this.props;
    if (navigator) {
      navigator.push({
        name: 'SigningUp',
        component: SigningUp,
      })
    }
 };
}

const styles = StyleSheet.create({
	scrollView:{
		height:height,
		width:width,
	},
	titleManifeste:{
		fontSize:30,
		marginTop:width*0.2,
		marginBottom:width*0.05,
		textAlign:'center',
	},
	manifesteText:{
		textAlign:'center',
	},
	manifesteRegister:{
		textAlign:'center',
		color: '#a4d147',
	},
});