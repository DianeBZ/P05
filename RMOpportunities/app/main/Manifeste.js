import React, { Component } from 'react';
import { Text, Image, StyleSheet, View, ScrollView, Dimensions, TouchableHighlight, BackAndroid, Navigator,} from 'react-native';

import Translation from './Translation';
import SigningUp from './SigningUp';
import LinearGradient from 'react-native-linear-gradient';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

// Page manifesteRegister
// You will see the main content in "ScrollView".
// _onPressRegister: go to the page of "SigningUp" 
// when you clicked "Registre".
export default class Manifeste extends Component{
   render(){
      BackAndroid.addEventListener('Back',this.onBackAndroid);
      return(
         <View style={styles.container}>
            <View style={styles.header} >
               <LinearGradient
                  start={{x: 0.0, y: 1.0}} end={{x: 1.0, y: 1.0}}
                  locations={[0,1.0]}
                  colors={['#185177', '#F6D092']}
                  style={styles.gradient}>
                  <View style={styles.headerIcon}>
                     <Image source={require('../img/recycle.png')} style = {styles.logoLeft} />
                  </View>
                  <Text style={styles.title}>
                     {Translation[lang].manifeste}
                  </Text>
                  <View style={styles.headerIcon}>
                     <Image source={require('../img/sprout.png')} style = {styles.logoRight} />
                  </View>
               </LinearGradient>
            </View>
            <ScrollView>
                <View style={styles.containerMain}>
                    <Text style={styles.manifesteText}>{Translation[lang].manifesteText}</Text>
                    <TouchableHighlight style={{marginTop:20, marginBottom:20}} onPress={this._onPressRegister} underlayColor='#F2F2F2'>
                       <Text style={styles.manifesteRegister}>{Translation[lang].manifesteRegister}</Text>
                    </TouchableHighlight>
                </View>
            </ScrollView>
         </View>
      );
   }
   onBackAndroid = () => {
      const { navigator } = this.props;
      if (navigator && navigator.getCurrentRoutes().length > 1) {
         BackAndroid.removeEventListener('Back', this.onBackAndroid);
         navigator.pop();
         return true;
      } else {
         return false;
      }
   };

   _onPressRegister = () => {
      BackAndroid.removeEventListener('Back', this.onBackAndroid);
      const { navigator } = this.props;
      if (navigator) {
         navigator.push({
            name: 'SigningUp',
            component: SigningUp,
         })
      }
   };
}

const styles = StyleSheet.create({
   container: {
      marginTop:height*0.08,
      flex:1,
   },
   gradient:{
      width: width,
      flexDirection:'row',
      paddingTop:15,
      paddingBottom:15
   },
   header: {
      alignItems: 'center',
      marginBottom:15,
   },
   headerIcon: {
      width:width/4,
      alignItems:'center',
      alignSelf:'center',
   },
   logoLeft:{
      transform: [{ rotate: '-25deg'}],
      width: 70,
      height: 70,
   },
   title: {
      width:width/2,
      fontSize: 30,
      textAlign: 'center',
      color: '#333333',
   },
   logoRight:{
      width: 70,
      height: 70,
   },
   containerMain: {
      width: width,
      justifyContent:'center',
      alignItems:'center',
      marginRight:5,
      marginBottom:15,
   },
   manifesteText:{
      textAlign:'center',
      color:"#000000"
   },
   manifesteRegister:{
      textAlign:'center',
      color: '#A4D04A',
      fontSize: 16,
   },
});
