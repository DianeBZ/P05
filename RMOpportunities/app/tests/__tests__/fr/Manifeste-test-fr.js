// __tests__/Manifeste-test-fr.js
import 'react-native';
import React from 'react';

global.lang = 'fr';
import Manifeste from '../../../main/Manifeste';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Manifeste />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
