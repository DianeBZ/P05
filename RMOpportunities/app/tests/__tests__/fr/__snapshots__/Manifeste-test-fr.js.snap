// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "marginTop": 106.72,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 15,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4279783799,
          4294365330,
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={
        Array [
          0,
          1,
        ]
      }
      start={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingTop": 15,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "width": 187.5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "transform": Array [
                Object {
                  "rotate": "-25deg",
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 30,
            "textAlign": "center",
            "width": 375,
          }
        }
      >
        Notre manifeste
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "width": 187.5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
      </View>
    </BVLinearGradient>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 15,
            "marginRight": 5,
            "width": 750,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "textAlign": "center",
            }
          }
        >
          Chez RMOpportunities, nous pensons que les ressources disponibles sur Terre sont trop importantes pour être gâchées. Nous croyons en la possibilité d’une deuxième vie pour un produit grâce notamment à l’économie circulaire qui rend l’utilisation des matières premières optimales grâce aux solutions que nous avons créées.
          						
          RMOpportunities offre d’un côté, une nouvelle façon aux entreprises de se libérer de leur stock de matières premières chimiques qui n’entrent plus dans le processus de production. Ceci permet d’améliorer leur empreinte écologique tout en réalisant d’importantes économies en évitant de devoir détruire des stocks encore utilisables. D'un autre côté nous aidons des entreprises en recherche de produits chimiques à s’en procurer à des prix très inférieurs au marché grâce à la valorisation de produits non utilisés. Notre ambition est d’offrir aux entreprises une politique zéro déchet pour leurs matières premières chimiques.
          						
          RMOpportunities se veut devenir le leader de la mise en relation pour la redistribution de produits chimiques en contribuant à la transformation de l’économie vers le modèle circulaire qui est écoresponsable.
          						
          Nous sommes motivés à l'idée de rendre le monde meilleur tout en générant des gains pour les entreprises. Si vous voulez acheter ou vendre des produits chimiques tout en protégeant l’environnement.
          
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "marginBottom": 20,
                "marginTop": 20,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#A4D04A",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Rejoignez-nous
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
